### PR TITLE
Search for language by country as well as name

### PR DESCRIPTION
**Pull request by JohnThomson**
Search for language by country as well as name

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/475)

### CODE REVIEW

File: SIL.WritingSystems/LanguageLookup.cs [hunk 1]
```diff
@@ -57,9 +58,59 @@ public LanguageLookup()
...
+						{
+							list = new List<LanguageInfo>();
+							_countryToLanguageIndex[country] = list;
+						}
+						list.Add(language);
+					}
+				}
+			}
+
+			var langData = LanguageRegistryResources.LanguageCodes.Replace("\r\n", "\n")
```
Comment (jasonleenaylor) [inline SIL.WritingSystems/LanguageLookup.cs]: This looks like it could be a problem on Linux

**Comment by megahirt** [comment]
Thanks for the work John.  Daniel Glassey is working on a response to your PR.  We'd like to see this logic of adding country to the search be a part of the index itself, as opposed to being bolted on after the index is already created.  The search facility has been simplified to just search an index, and not generate the index (this is now done separately).

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-:-Khz2CFSkkigRjQIVFAH:b-lm0u0s)*

**Comment by megahirt** [comment]
just noticed that this PR includes the changes from PR 474 as well.  Can these be separated?  Not sure if closing and re-opening PRs is the best approach or not.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-:-Khz3d6KH-P8pf-26n5O:b-1ttb6v)*

**Comment by glasseyes** [comment]
comments in PR474

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[SIL.WritingSystems/LanguageLookup.cs, line 80 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-KhxaYw4knSnYvJ7YgJe-r1-80:-Khz3kdCaAV3bnSglpjE:bgt5gnr) ([raw file](https://github.com/sillsdev/libpalaso/blob/88b30b4c9574ebb9effc58996f2c5e57be74f7ee/SIL.WritingSystems/LanguageLookup.cs#L80)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

This looks like it could be a problem on Linux
</blockquote></details>

That replace and split has been used a lot for reading in the data files and making sure they are the same on windows and linux (this part is in PR474)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-:-KhzAFopJtn_wdEGCwcA:b-3ni1lx)*

**Comment by JohnThomson** [comment]
Will rebase at some point. It's hard to work on multiple things in the same file when it takes time to get reviews.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[SIL.WritingSystems/LanguageLookup.cs, line 80 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-KhxaYw4knSnYvJ7YgJe-r1-80:-Ki0SUAfpwq2x409Vp_F:b8p6wim) ([raw file](https://github.com/sillsdev/libpalaso/blob/88b30b4c9574ebb9effc58996f2c5e57be74f7ee/SIL.WritingSystems/LanguageLookup.cs#L80)):*
<details><summary><i>Previously, glasseyes wrote…</i></summary><blockquote>

That replace and split has been used a lot for reading in the data files and making sure they are the same on windows and linux (this part is in PR474)
</blockquote></details>

I don't see the problem. We're not writing this data out, we're just making the read code not care whether we get Windows or Linux line breaks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-:-Ki0SfBtlljErTABoLOd:b-t5yudc)*

**Comment by JohnThomson** [comment]
Don't merge this. Needs rework based on final version of https://github.com/sillsdev/libpalaso/pull/478

**Comment by jasonleenaylor** [comment]
*[SIL.WritingSystems/LanguageLookup.cs, line 80 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/475#-KhxaYw4knSnYvJ7YgJe-r1-80:-Ki_6hcDpoDc6MsKaAPS:b-6bb3oo) ([raw file](https://github.com/sillsdev/libpalaso/blob/88b30b4c9574ebb9effc58996f2c5e57be74f7ee/SIL.WritingSystems/LanguageLookup.cs#L80)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

I don't see the problem. We're not writing this data out, we're just making the read code not care whether we get Windows or Linux line breaks.
</blockquote></details>

Yes, it was late on Friday. Withdrawn.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/475)*

**Comment by JohnThomson** [comment]
Abandoned in favor of https://github.com/sillsdev/libpalaso/pull/489

**Pull request closed without merging**